File: solutions/csharp/leap/1/Leap.cs
public static class Leap
{
    public static bool IsLeapYear(int year)
    {
        bool IsMultipleOf(int divisor) => year % divisor == 0;

        return IsMultipleOf(4) && (!IsMultipleOf(100) || IsMultipleOf(400));
    }
}

File: solutions/java/leap/1/src/main/java/Leap.java
class Leap {

    boolean isLeapYear(int year) {
        return isMultipleOf(year, 4)
                && (!isMultipleOf(year, 100) || isMultipleOf(year, 400));
    }

    private boolean isMultipleOf(int year, int factor) {
        return year % factor == 0;
    }

}
